pkg/network/amqp.go

```diff
@@ -71,6 +72,77 @@ func (a *Amqp) Start(started chan bool) {
 	started <- true
 }
 
+// DeclareQueue declare a queue in amqp handler
+func (a *Amqp) DeclareQueue(queueName, exchangeName string) error {
+	err := a.channel.ExchangeDeclare(
+		exchangeName,
+		amqp.ExchangeTopic, // type
+		true,               // durable
+		false,              // delete when complete
+		false,              // internal
+		false,              // noWait
+		nil,                // arguments
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	queue, err := a.channel.QueueDeclare(
+		queueName,
+		true,  // durable
+		false, // delete when unused
+		false, // exclusive
+		false, // noWait
+		nil,   // arguments
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	a.queue = &queue
+	return nil
+}
+
+// OnMessage receive messages an put them on channel
+func (a *Amqp) OnMessage(msgChan chan InMsg, queueName, exchange, key string) error {
+
+	err := a.channel.QueueBind(
+		queueName,
+		key,
+		exchange,
+		false,
+		nil,
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	deliveries, err := a.channel.Consume(
+		queueName,
+		"",    // consumerTag
+		true,  // noAck
+		false, // exclusive
+		false, // noLocal
+		false, // noWait
+		nil,   // arguments
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	go func(deliveries <-chan amqp.Delivery) {
```

Comment:
One option to this anonymous function is to move it to another method (specific to delivery handling).

```
go handle(deliveries)

return nil
```

Reply:
I chose to use an anonymous function because the variable msgChan is this scope so I could avoid to pass it as a parameter.

I changed to `convertDeliveryToInMsg` because `handle` gives me the idea that I already handle the message but I just pass the message into a channel to be processed to other methods that will handle the message